src/lib.rs

```diff
@@ -54,13 +48,20 @@ where
     ) -> Poll<Option<Self::Item>> {
         let s = Pin::into_inner(self);
 
-        Pin::new(&mut s.stream).poll_data(cx).map(|f| match f {
-            Some(Ok(b)) => {
-                let slice = b.chunk();
-                let a = Uint8Array::new(&JsValue::from(slice.len()));
-                a.copy_from(slice);
-                let out: JsValue = a.into();
-                Some(Ok(out))
+        Pin::new(&mut s.stream).poll_frame(cx).map(|f| match f {
+            Some(Ok(frame)) => {
+                match frame.into_data() {
+                    Ok(b) => {
+                        let slice = b.chunk();
+                        let a = Uint8Array::new(&JsValue::from(slice.len()));
+                        a.copy_from(slice);
+                        let out: JsValue = a.into();
+                        Some(Ok(out))
+                    }
+                    // Not 100% sure how to best handle this error besides informing the user.
+                    // This gives a frame that isn't a data frame and I don't know how that could be useful here?
+                    Err(_) => Some(Err(JsValue::from_str("Not a data frame"))),
```

Comment:
Good comment, I think this behaviour is fine for now

Reply:
Thanks!